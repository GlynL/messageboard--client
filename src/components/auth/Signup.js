import React, { Component } from "react";
import { connect } from "react-redux";
import * as actions from "../../actions/auth";

import Form from "./Form";

export class Signup extends Component {
  handleSubmit = user => {
    this.props.signup(user, () => this.props.history.push("/"));
  };

  render() {
    return (
      <div>
        <h1>Signup</h1>
        {this.props.errorMessage && <div>{this.props.errorMessage}</div>}
        <Form handleSubmit={this.handleSubmit} />
      </div>
    );
  }
}

const mapStateToProps = state => ({ errorMessage: state.auth.errorMessage });

export default connect(
  mapStateToProps,
  actions
)(Signup);
