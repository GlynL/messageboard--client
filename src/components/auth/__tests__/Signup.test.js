import React from "react";
import { shallow } from "enzyme";
import { Signup } from "../Signup";

describe("w/out props", () => {
  let wrapper, history, signup;
  beforeEach(() => {
    history = { push: jest.fn() };
    signup = jest.fn();
    wrapper = shallow(<Signup history={history} signup={signup} />);
  });

  it("renders Signup correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

it("renders Signup with error prop correctly", () => {
  const wrapper = shallow(<Signup errorMessage="test error" />);
  expect(wrapper).toMatchSnapshot();
});
