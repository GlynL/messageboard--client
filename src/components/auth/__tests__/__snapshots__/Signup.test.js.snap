// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Signup with error prop correctly 1`] = `
<div>
  <h1>
    Signup
  </h1>
  <div>
    test error
  </div>
  <Form
    buttonText="Sign Up"
    handleSubmit={[Function]}
  />
</div>
`;

exports[`w/out props renders Signup correctly 1`] = `
<div>
  <h1>
    Signup
  </h1>
  <Form
    buttonText="Sign Up"
    handleSubmit={[Function]}
  />
</div>
`;
